index.js

```diff
@@ -154,22 +154,31 @@ module.exports = class ServeDrive extends ReadyResource {
     const id = searchParams.get('drive') // String or null
     const filename = decodeURI(pathname)
 
-    const drive = await this.getDrive(id, filename)
+    let drive = null
+    let error = null
 
     try {
+      drive = await this.getDrive(id, filename)
       await this._driveToRequest(drive, req, res, filename, id, version)
     } catch (e) {
-      safetyCatch(e)
-      this.emit('request-error', e)
+      error = e
+    }
 
-      if (!res.headersSent) {
-        res.writeHead(500)
-        const msg = e.code || e.message
-        res.end(msg)
-      }
-    } finally {
-      await this.releaseDrive(id)
+    try {
+      if (drive !== null) await this.releaseDrive(id)
+    } catch (e) {
+      error = e
     }
+
+    if (error === null) return
+
+    if (!res.headersSent) {
+      res.writeHead(500, { 'Content-Type': 'text/plain' })
+      const msg = error.code || error.message
+      res.end(msg)
+    }
+
+    this.emit('request-error', error)
```

Comment:
Would put this right after `if (error === null) return` , that way it will always emit if there's an error (we no longer rely on the code between that line and this one being error-free)

Reply:
imo emits should always come at the end, as you are giving up control flow. ie, someone might close from this.